core/src/main/java/org/apache/brooklyn/util/core/logbook/opensearch/OpenSearchLogStore.java

```diff
@@ -253,17 +250,47 @@ protected String getJSONQuery(LogBookQueryParams params) {
             queryBoolMustListBuilder.add(ImmutableMap.of("range", ImmutableMap.of("timestamp", timestampMapBuilder.build())));
         }
 
+        // Apply search taskId.
+        if (Strings.isNonBlank(params.getTaskId())) {
+            queryBoolMustListBuilder.add(
+                    ImmutableMap.of("bool",
+                            ImmutableMap.of("should",
```

Comment:
`should` is an OR match condition, meaning to look for taskId in the `taskId` filed or `message` field, right?

Reply:
Short answer: yes, it how it work when it's inside a "bool" sentence. The docs are here https://www.elastic.co/guide/en/elasticsearch/reference/6.8/query-dsl-bool-query.html but I needed to see some examples to have the query working